### PR TITLE
[Decoder] update callback description

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Update callback description for decoder sub-plugin.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/nnstreamer_plugin_api_decoder.h [hunk 1]
```diff
@@ -40,26 +40,60 @@ typedef struct _GstTensorDecoderDef
   char *modename;
       /**< Unique decoder name. GST users choose decoders with mode="modename". */
   int (*init) (void **private_data);
-      /**< Object initialization for the decoder */
+      /**< Object initialization for the decoder.
+       *
+       * @param[in/out] private_data A sub-plugin may save its internal private data here. The sub-plugin is responsible for alloc/free of this pointer.
```
Comment (myungjoo) [inline gst/nnstreamer/nnstreamer_plugin_api_decoder.h]: To clarify, let's add: ```Normally, the sub-plugin may allocate the private_data with this function```

File: gst/nnstreamer/nnstreamer_plugin_api_decoder.h [hunk 2]
```diff
@@ -40,26 +40,60 @@ typedef struct _GstTensorDecoderDef
...
+       * @return TRUE if OK. FALSE if error.
+       */
   void (*exit) (void **private_data);
-      /**< Object destruction for the decoder */
+      /**< Object destruction for the decoder.
+       *
+       * @param[in/out] private_data A sub-plugin may save its internal private data here. The sub-plugin is responsible for alloc/free of this pointer.
```
Comment (myungjoo) [inline gst/nnstreamer/nnstreamer_plugin_api_decoder.h]: Again, to clarify, let's add: ```Normally, the sub-plugin may free the private_data with this function.```

File: gst/nnstreamer/nnstreamer_plugin_api_decoder.h [hunk 3]
```diff
@@ -40,26 +40,60 @@ typedef struct _GstTensorDecoderDef
...
-        * Note that we support single-tensor (other/tensor) only!
-        * @todo WIP: support multi-tensor for input!!!
-        */
+      /**< The function to be called when the input tensor incomes into tensor_decoder.
+       * The sub-plugin should update the output buffer. outbuf must be allocated but empty (gst_buffer_get_size (outbuf) == 0).
+       *
+       * @param[in/out] private_data A sub-plugin may save its internal private data here. The sub-plugin is responsible for alloc/free of this pointer.
+       * @param[in] config The structure of input tensor info.
+       * @param[in] input The array of input tensor data. The maximum array size of input data is NNS_TENSOR_SIZE_LIMIT.
+       * @param[in/out] outbuf A sub-plugin should update or append proper memory for the negotiated media type.
```
Comment (myungjoo) [inline gst/nnstreamer/nnstreamer_plugin_api_decoder.h]: Isn't this [out] ?

Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_plugin_api_decoder.h]: you are right, updated.

File: gst/nnstreamer/nnstreamer_plugin_api_decoder.h (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_plugin_api_decoder.h]: I updated your comment.

**Pull request merged**